spec/models/grda_warehouse/cohort_spec.rb

```diff
@@ -134,4 +146,49 @@
       expect(updated_collection.name).to eq('New Name')
     end
   end
+
+  describe 'column_state handling' do
+    let!(:test_column) { build :user_string_cohort_column_1 }
+
+    before do
+      # Add test column to cohort's column_state
+      columns = cohort.column_state || []
+      test_column.cohort_column.activate
+      columns << test_column
+      cohort.update(column_state: columns)
```

Comment:
Do you think there's any benefit to testing this against `GrdaWarehouse::Cohorts::CohortColumn.known_cohort_columns`?  
I _think_ that would enforce that all of those columns can be instantiated (and have classes that back them).  Basically:
 ```suggestion
      @all_columns = GrdaWarehouse::Cohort.available_columns
      @all_columns.each { |c| c.cohort_column.activate }
      GrdaWarehouse::Cohorts::CohortColumn.known_cohort_columns
      columns = cohort.column_state || @all_columns
      cohort.update(column_state: columns)
```

Reply:
I like it, and worst comes we should have a test checking that the classes in the database also exist in the code. I'll give it a run locally and see what falls out